### PR TITLE
Rl backup solutions - rsnapshot

### PR DESCRIPTION
**Pull request by sspencerwire**
This is the first document for submitted for backup solutions for Rocky Linux

### CODE REVIEW

**Comment by sspencerwire** [comment]
Let me know if it needs revision

File: en/rocky/8/guides/RL_rsnapshot_backup.md [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+We are going to set our backup to automatically run at 11 PM, so we will add this to the crontab: 
+
+```
+## Running the backup at 11 PM
+00 23 *  *  *  /usr/bin/rsnapshot -c /etc/rsnapshot.conf beta`
+```
+
+## Multiple Machine or Multiple Server Backups
+
+Doing backups of multiple machines from a machine with a RAID array or large storage capacity, either on premises or from across the Internet works very well. If running from across the Internet, you need to make sure that both locations have adequate bandwidth for the backups to occur. In my previous job, we used both an on-premise disk array and an off-premise disk array for maximum redundancy.
```
Comment (sristudentnetwork) [inline en/rocky/8/guides/RL_rsnapshot_backup.md:194]: Maybe rephrase the last sentence to something like: "To maximize redundancy, you can use Rsnapshot to synchronize an on-site storage volume and an off-site backup array."?

File: en/rocky/8/guides/RL_rsnapshot_backup.md [hunk 2]
```diff
@@ -0,0 +1,361 @@
+# Rocky Linux - Backup Solution - Rsnapshot
+
+# Introduction
+
+Rsnapshot is a very powerful backup utility that can be installed on any Linux-based machine. It can either backup a machine locally, or you can backup multiple machines, say servers for instance, from a single machine. Rsnapshot uses rsync and is written entirely in perl with no library dependencies, so there are no weird requirements to installing it. In the case of Rocky Linux, you should be able to install rsnapshot simply by installing the EPEL repository. This documentation covers the installation of rsnapshot on Rocky Linux only.
+
+## Prerequisites
```
Comment (calderds) [inline en/rocky/8/guides/RL_rsnapshot_backup.md:7]: (Maybe a style guide issue) Maybe have a sentence like "To successfully understand and implement the instructions in this documentation, we assume you already:"

File: en/rocky/8/guides/RL_rsnapshot_backup.md [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+========================================================================================================================================
+ Package                           Architecture                 Version                              Repository                    Size
+========================================================================================================================================
+Installing:
+ rsnapshot                         noarch                       1.4.3-1.el8                          epel                         121 k
+Installing dependencies:
+ perl-Lchown                       x86_64                       1.01-14.el8                          epel                          18 k
+ rsync                             x86_64                       3.1.3-9.el8                          baseos                       404 k
+
+Transaction SummaryIn this case, rsnapshot is going to be run locally to backup a particular machine. For this example,  the configuration file will be broken down for specific things that need to be changed in the file.
```
Comment (calderds) [inline en/rocky/8/guides/RL_rsnapshot_backup.md:49]: I really like this example for package installation. Is this Transaction Summary line supposed to be here?

File: en/rocky/8/guides/RL_rsnapshot_backup.md [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+
+Total download size: 543 k
+Installed size: 1.2 M
+Is this ok [y/N]: y
+```
+## Mounting A Drive or Filesystem For Backup
+
+This particular step is only necessary if you are backing up a single machine or server (our first example below).
+
+For this particular step, we are assuming an external hard drive of some sort that will be attached to the USB port on your machine, which would likely be the most common method of backing up a single machine.
```
Comment (calderds) [inline en/rocky/8/guides/RL_rsnapshot_backup.md:61]: Maybe combine this sentence and the previous one to describe what's going on in this section:
"""
In this step, we show how to mount a hard drive, such as an external USB hard drive, that will be used to back up your system. This particular step is only necessary if you are backing up a single machine or server, as seen in our first example below.
"""

File: en/rocky/8/guides/RL_rsnapshot_backup.md [hunk 5]
```diff
@@ -0,0 +1,361 @@
...
+Installed size: 1.2 M
+Is this ok [y/N]: y
+```
+## Mounting A Drive or Filesystem For Backup
+
+This particular step is only necessary if you are backing up a single machine or server (our first example below).
+
+For this particular step, we are assuming an external hard drive of some sort that will be attached to the USB port on your machine, which would likely be the most common method of backing up a single machine. 
+
+1. Plugin the USB drive.
```
Comment (calderds) [inline en/rocky/8/guides/RL_rsnapshot_backup.md:63]: Plug in

File: en/rocky/8/guides/RL_rsnapshot_backup.md [hunk 6]
```diff
@@ -0,0 +1,361 @@
...
+7. Now type `mount | grep sda1` again, and you should see something like this: `/dev/sda1 on /mnt/backup type ext2 (rw,relatime)`
+8. Next create a directory that must exist for the backup to continue on the mounted drive. We are using storage for this example: `sudo mkdir /mnt/backup/storage`
+
+Note that for a single machine, you will have to either repeat the umount and mount steps each time the drive is plugged in again or each time the system reboots, or automate these commands with a script.
+
+# Configuring Rsnapshot
+
+This is the most important step. One of the easiest mistakes to make here is to put spaces between things. The rsnapshot configuration requires tabs for any separation between elements. A warning to that effect is at the very top of the configuration file. A space entered somewhere will cause the entire configuration-and your backup-to fail. The best thing is that the default configuration that comes with rsnapshot only needs minor changes to make it work for a backup of a local machine. It's always a good idea, though, to make a backup copy of the configuration file before you start editing:
```
Comment (calderds) [inline en/rocky/8/guides/RL_rsnapshot_backup.md:76]: "spaces between things" - can we be more specific here? I know you get into more detail later, but it might be helpful here.

File: en/rocky/8/guides/RL_rsnapshot_backup.md [hunk 7]
```diff
@@ -0,0 +1,361 @@
...
+
+### Setting The Backup To Run Automatically
+
+Once everything has been tested and we know that things will work without issue, the next step is to set up the crontab for root's so that all of this can be done automatically every day:
+
+`sudo crontab -e`
+
+If you haven't run this before, choose vim.basic as your editor or your own editor preference when the `Select an editor` line comes up.
```
Comment (calderds) [inline en/rocky/8/guides/RL_rsnapshot_backup.md:183]: Should we mention something about environment variables here or in the prerequisites? Maybe in the bash scripting part...

File: en/rocky/8/guides/RL_rsnapshot_backup.md [hunk 8]
```diff
@@ -0,0 +1,361 @@
...
+Doing backups of multiple machines from a machine with a RAID array or large storage capacity, either on premises or from across the Internet works very well. If running from across the Internet, you need to make sure that both locations have adequate bandwidth for the backups to occur. In my previous job, we used both an on-premise disk array and an off-premise disk array for maximum redundancy. 
+
+### Assumptions
+
+We are assuming that you are running rsnapshot from a machine remotely, on-premise. This exact configuration can be duplicated, as indicated above, remotely off-premise as well.
+
+In this case, you will want to install rsnapshot on the machine that is doing all of the backups. You'll also need to:
+
+* make sure that the servers you will be backing up to, have a firewall rule that allows the remote machine to SSH into it
+* that each server that you will be backing up has a recent version of rsync installed. For Rocky Linux servers, you need only run `dnf install rsync` to make sure you are running a recent version.
```
Comment (calderds) [inline en/rocky/8/guides/RL_rsnapshot_backup.md:203]: We can simplify the second sentence: For Rocky Linux servers, run `dnf install rsync` to update your system's version of rsync.

File: en/rocky/8/guides/RL_rsnapshot_backup.md [hunk 9]
```diff
@@ -0,0 +1,361 @@
...
+Once you have verified that you can SSH in without a password, remove the id_rsa.pub file from the /root directory on each machine. 
+
+`rm id_rsa.pub`
+
+#### SSH Directory and authorized_keys Security
+
+On each of your target machines, make sure that the following permissions are applied:
+
+`chmod 700 .ssh/`
+`chmod 600 .ssh/authorized_keys`
```
Comment (calderds) [inline en/rocky/8/guides/RL_rsnapshot_backup.md:275]: This is really good! I think we should make it its own document and reference it from here - this sort of procedure is likely important for other applications as well.

File: en/rocky/8/guides/RL_rsnapshot_backup.md [hunk 10]
```diff
@@ -0,0 +1,361 @@
...
+
+And
+
+`no_create_root 1`
+
+can be remarked out again
+
+`#no_create_root 1`
+
+The other difference here is that each machine will have its very own configuration. You'll also want to remove the Once you get used to this, you'll simply copy one of your existing configuration files over to a new name and then modify it to fit any additional machines that you want to backup. For now, we want to modify the configuration just like we did above, and then save it. Then copy that file as a template for our first server:
```
Comment (calderds) [inline en/rocky/8/guides/RL_rsnapshot_backup.md:293]: Typo in second sentence

File: en/rocky/8/guides/RL_rsnapshot_backup.md (hunk 1 above)
Comment (calderds) [inline en/rocky/8/guides/RL_rsnapshot_backup.md:194]: Maybe revise the last sentence to: "You can use rsnapshot to synchronize an on-site server with an off-site backup array or backup server to improve data redundancy."

**Comment by calderds** [comment]
Great job! This is really good stuff. A couple minor changes, but it looks good overall on the writing side.

 We should figure out a staging area to place documents for them to be reviewed for technical accuracy. Maybe a branch called `staging` that we could merge into once the writing is solid.
